### PR TITLE
added order controllers; admin controller for products

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

**Review by Victoria-Moiseienko — COMMENTED** [review]
Good job!

File: src/main/java/com/internet/shop/controllers/DeleteProductFromShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        Long productID = Long.valueOf(req.getParameter("id"));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/DeleteProductFromShoppingCartController.java]:
```suggestion
        Long productId = Long.valueOf(req.getParameter("id"));
```

File: src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllProductsAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java]:
```suggestion
public class ManageProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/DeleteProductController.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long productId = Long.valueOf(req.getParameter("id"));
+        productService.delete(productId);
+        resp.sendRedirect(req.getContextPath() + "/product/all/admin");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/DeleteProductController.java]: let's do the same after we add a new product to storage in AddProductController: `resp.sendRedirect(req.getContextPath() + "/product/all/admin")`

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 4]
```diff
@@ -6,11 +6,13 @@
 <body>
 <h1>Hello World</h1>
 <h1>Welcome</h1>
-<a href="${pageContext.request.contextPath}/injectData">inject test data</a>
-<a href="${pageContext.request.contextPath}/user/all">user/all</a>
-<a href="${pageContext.request.contextPath}/registration">register</a>
-<a href="${pageContext.request.contextPath}/product/all">all products</a>
-<a href="${pageContext.request.contextPath}/product/add">add new product</a>
-<a href="${pageContext.request.contextPath}/cart/products">cart</a>
+<a href="${pageContext.request.contextPath}/injectData">inject test data</a><br>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/index.jsp]:
```suggestion
<a href="${pageContext.request.contextPath}/inject-data">inject test data</a><br>
```

File: src/main/webapp/web.xml [hunk 5]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createOrder</servlet-name>
+        <servlet-class>com.internet.shop.controllers.CreateOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>createOrder</servlet-name>
+        <url-pattern>/order/add</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/complete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 6]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>userOrders</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetUserOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>userOrders</servlet-name>
+        <url-pattern>/order/users</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>orderProducts</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetOrderDetailsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>orderProducts</servlet-name>
+        <url-pattern>/order/details</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/details</url-pattern>
```

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteOrder</servlet-name>
+        <servlet-class>com.internet.shop.controllers.DeleteOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteOrder</servlet-name>
+        <url-pattern>/order/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 9]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allOrders</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetAllOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allOrders</servlet-name>
+        <url-pattern>/order/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/all</url-pattern>
```

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetAllProductsAdminController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <url-pattern>/product/all/admin</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -92,4 +92,84 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProduct</servlet-name>
+        <servlet-class>com.internet.shop.controllers.DeleteProductController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteProduct</servlet-name>
+        <url-pattern>/product/all/admin/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/delete</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/DeleteProductFromShoppingCartController.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/DeleteProductFromShoppingCartController.java]: done

File: src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/DeleteProductController.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/DeleteProductController.java]: done

File: src/main/webapp/WEB-INF/views/index.jsp (hunk 4 above)
Comment (TimoxaV) [reply src/main/webapp/WEB-INF/views/index.jsp]: done

File: src/main/webapp/web.xml (hunk 5 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 6 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 7 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 8 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 9 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 10 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 11 above)
Comment (TimoxaV) [reply src/main/webapp/web.xml]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**